### ISSUE TITLE
log file grows to quarter gig in <10 minutes, still no response from listpeers or listchannels

### ISSUE THREAD

**Issue by Ulmo** [ElementsProject/lightning#826]
Subject says it all.  I compiled latest software, stopped existing dragon and restarted lightningd, and the log file grew to a quarter gigabyte in less than 10 minutes, and it still won't respond to lightning-cli listpeers or lightning-cli listchannels, so I have no idea what the state of it is.

I tried reverting to a version from two days ago, and the same thing happened, so I reverted the reversion back to the newest, since if it's going to go megamad, might as well be current so we can debug.

Something just happened:  one of the hundred or so "connect" commands I issued responded with one connection being opened.  listpeers and listchannels still don't work.

Something odd happened the first time and every time since today that I restarted it:  24593297 Satoshis appeared in my wallet.  I'm not quite sure where they were before and why they weren't there before.  It always feels good to receive unknown money, but it causes me to wonder where it's been and what else might be missing.

Is it really true I can't even know the balance of my own channels?

Update:  a few dozen channels just came through as connected.  Ok, dozens more ... it seems that when the logfile gets to around size 259881712, some things start working again.  But, listpeers & listchannels still don't work.

**Comment by Ulmo** [comment]
Here's the ps axf output regarding lightningd:

18677 pts/7    S      3:09  |   \_ lightningd --network bitcoin --log-level=debug --ignore-fee-limits=false --log-file=/home/ulmo/log/log --alias=soonandsoforth
18678 pts/7    Z      0:00  |       \_ [lightning_chann] <defunct>
18679 pts/7    Z      0:00  |       \_ [lightning_closi] <defunct>
18680 pts/7    Z      0:00  |       \_ [lightning_gossi] <defunct>
18681 pts/7    Z      0:00  |       \_ [lightning_hsmd] <defunct>
18682 pts/7    Z      0:00  |       \_ [lightning_oncha] <defunct>
18683 pts/7    Z      0:00  |       \_ [lightning_openi] <defunct>
18684 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_hsmd
18685 pts/7    R     12:35  |       \_ /usr/local/libexec/c-lightning/lightning_gossipd
18695 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
18696 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
18697 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
18700 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
18705 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
18707 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
18708 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
18710 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
18718 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
18787 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
18796 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
18811 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
18814 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
18821 pts/7    S      0:01  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
18827 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
18834 pts/7    S      0:01  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
18835 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
18844 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
18848 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
18858 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
18876 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
18891 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
18894 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
18907 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
18921 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
18939 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
18981 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
19108 pts/7    S      0:01  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
19226 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
19403 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
19455 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
24163 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
24891 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld
26003 pts/7    S      0:00  |       \_ /usr/local/libexec/c-lightning/lightning_channeld

For reference, here's a similar node's ps axf output for lightningd, still running on 2 day old software:

14787 pts/10   S     39:03  |                   \_ lightningd --network=bitcoin --log-level=debug --ignore-fee-limits=false --ipaddr=1.2.3.4 --log-file=/home/ulmo/logs/log --alias=anotherone
14788 pts/10   Z      0:00  |                   |   \_ [lightning_chann] <defunct>
14789 pts/10   Z      0:00  |                   |   \_ [lightning_closi] <defunct>
14790 pts/10   Z      0:00  |                   |   \_ [lightning_gossi] <defunct>
14791 pts/10   Z      0:00  |                   |   \_ [lightning_hsmd] <defunct>
14792 pts/10   Z      0:00  |                   |   \_ [lightning_oncha] <defunct>
14793 pts/10   Z      0:00  |                   |   \_ [lightning_openi] <defunct>
14802 pts/10   S      0:00  |                   |   \_ /usr/local/libexec/c-lightning/lightning_hsmd
14813 pts/10   S      5:27  |                   |   \_ /usr/local/libexec/c-lightning/lightning_gossipd
15982 pts/10   S      0:03  |                   |   \_ /usr/local/libexec/c-lightning/lightning_channeld
 6893 pts/10   S      0:03  |                   |   \_ /usr/local/libexec/c-lightning/lightning_channeld
 9658 pts/10   S      0:00  |                   |   \_ /usr/local/libexec/c-lightning/lightning_channeld
 2353 pts/10   S      0:00  |                   |   \_ /usr/local/libexec/c-lightning/lightning_onchaind
15938 pts/10   S      0:01  |                   |   \_ /usr/local/libexec/c-lightning/lightning_channeld
20854 pts/10   S      0:01  |                   |   \_ /usr/local/libexec/c-lightning/lightning_channeld
15704 pts/10   S      0:00  |                   |   \_ /usr/local/libexec/c-lightning/lightning_channeld
12516 pts/10   S      0:00  |                   |   \_ /usr/local/libexec/c-lightning/lightning_channeld
15151 pts/10   S      0:01  |                   |   \_ /usr/local/libexec/c-lightning/lightning_channeld
30915 pts/10   S      0:02  |                   |   \_ /usr/local/libexec/c-lightning/lightning_channeld
28313 pts/10   S      0:01  |                   |   \_ /usr/local/libexec/c-lightning/lightning_channeld
14172 pts/10   S      4:06  |                   |   \_ /usr/local/libexec/c-lightning/lightning_channeld
  589 pts/10   S      0:01  |                   |   \_ /usr/local/libexec/c-lightning/lightning_channeld
  590 pts/10   S      0:01  |                   |   \_ /usr/local/libexec/c-lightning/lightning_channeld
  611 pts/10   S      0:01  |                   |   \_ /usr/local/libexec/c-lightning/lightning_channeld
19067 pts/10   S      0:01  |                   |   \_ /usr/local/libexec/c-lightning/lightning_channeld
14031 pts/10   S      0:01  |                   |   \_ /usr/local/libexec/c-lightning/lightning_channeld
 4356 pts/10   S      1:53  |                   |   \_ /usr/local/libexec/c-lightning/lightning_channeld
31659 pts/10   S      0:00  |                   |   \_ /usr/local/libexec/c-lightning/lightning_onchaind
24403 pts/10   S      0:00  |                   |   \_ /usr/local/libexec/c-lightning/lightning_channeld
 6970 pts/10   S      0:00  |                   |   \_ /usr/local/libexec/c-lightning/lightning_channeld
10678 pts/10   S      0:00  |                   |   \_ /usr/local/libexec/c-lightning/lightning_channeld
11300 pts/10   S      0:01  |                   |   \_ /usr/local/libexec/c-lightning/lightning_channeld

Note that on that node lightningd has run cpu for 39 minutes, but gossipd only 5 minutes in two days, as opposed to the new software on my other node gossipd getting to 13 minutes cpu (and climbing) in only a quarter hour.  Now at 16:53.

There's plenty of memory:  3/4s is buffers, so it's not that.

**Comment by Ulmo** [comment]
Here's a sample of its output:

lightning_gossipd(18685): TRACE: Deferring update for pending channel 505627:2473:1(0)
lightning_gossipd(18685): TRACE: Received channel_update for channel 506204:1349:1(0)
lightning_gossipd(18685): TRACE: Replacing existing update
lightning_gossipd(18685): TRACE: Deferring update for pending channel 506204:1349:1(0)
lightning_gossipd(18685): TRACE: Received channel_update for channel 505863:711:0(0)
lightning_gossipd(18685): TRACE: Replacing existing update
lightning_gossipd(18685): TRACE: Deferring update for pending channel 505863:711:0(0)
lightning_gossipd(18685): TRACE: Received channel_update for channel 506100:1846:0(1)
lightning_gossipd(18685): TRACE: Replacing existing update
lightning_gossipd(18685): TRACE: Deferring update for pending channel 506100:1846:0(1)
lightning_gossipd(18685): TRACE: Received node_announcement for node 0239e5323f33ce3729f69ddd08760e83a9d6729521da95f28b93dadfb4d0b26ad9
lightning_gossipd(18685): TRACE: Node not found, was the node_announcement preceded by at least channel_announcement?
lightning_gossipd(18685): TRACE: Received channel_announcement for channel 506100:1776:1
lightning_gossipd(18685): UPDATE WIRE_GOSSIP_GET_TXOUT
lightning_gossipd(18685): TRACE: Received channel_update for channel 505863:711:0(1)
lightning_gossipd(18685): TRACE: Replacing existing update
lightning_gossipd(18685): TRACE: Deferring update for pending channel 505863:711:0(1)
lightning_gossipd(18685): TRACE: Received channel_announcement for channel 506469:1050:0
lightning_gossipd(18685): UPDATE WIRE_GOSSIP_GET_TXOUT
lightning_gossipd(18685): TRACE: Received channel_update for channel 505357:1761:1(0)
lightning_gossipd(18685): TRACE: Replacing existing update
lightning_gossipd(18685): TRACE: Deferring update for pending channel 505357:1761:1(0)
lightning_gossipd(18685): TRACE: Received node_announcement for node 023d280ae29f84dcfd289eb66b57227fea3a7bde97ec2869923d3c4d5c6a426028
lightning_gossipd(18685): TRACE: Node not found, was the node_announcement preceded by at least channel_announcement?

I ran a few simple commands to analyze that:

$ egrep 18685 log|sort -u|wc -l
6355
$ egrep 18685 log|sort |wc -l
266212

**Comment by Ulmo** [comment]
Well, it's not just that process; that's only a tenth of the log file.  In all, there are a lot of non-unique log entries:

$ sort -u log | wc -l
161993
$ wc -l log
1218338

Everything is getting repeated ten times.

**Comment by Ulmo** [comment]
The lightning.sqlite3 database size has remained relatively constant, at only 106,496 bytes.  I suppose it's nice the database isn't getting ruined.  Its last file mod time was 11 minutes ago at 18:31, and lightningd started at 18:15, so whatever is happening recently but not very recently updated the database.

**Comment by Ulmo** [comment]
fundchannel also hangs.

**Comment by Ulmo** [comment]
getroute also hangs.

**Comment by Ulmo** [comment]
Still growing.  Still listpeers, listchannels, getroute still hang.  I didn't test fundchannel or pay, but they're still sitting there hung from before.

Looking at https://lnmainnet.gaben.win I do see my node listed, so perhaps it's running fully.  But what is it really doing, and how can I diagnose it from the computer running it?  Must I install a separate node to monitor it, or some sort of other software?

**Comment by Ulmo** [comment]
New news!  listpeers, listchannels working.  pay responds.  fundchannel responds.  What shook loose?

**Comment by Ulmo** [comment]
FYI, this is how much CPU before it unclogged:

18677 pts/7    R     23:36  |   \_ lightningd
18685 pts/7    S     85:43  |       \_ /usr/local/libexec/c-lightning/lightning_gossipd

**Comment by Ulmo** [comment]
Three new developments:

1.  The .25BTC are gone again.
2.  The host disconnected my SSH session into it.  I'm not sure why.  When I logged back in, the entire session had ended.  I don't think I hit a kill command.  But, lightningd and its processes are gone.

Ok, that's annoying, to say the least.  I have to restart it again.

**Comment by Ulmo** [comment]
When I restarted it again, the .25BTC reappear.

I'm going to go do something else.  Hopefully this thing stops misbehaving.

**Comment by RvMp** [comment]
See: https://github.com/ElementsProject/lightning/issues/796

**Comment by cdecker** [comment]
The logfile growth with `--log-level=debug` is to be expected, but it'll settle down to a slower growth once the initial gossip sync is done. c-lightning currently is also not optimized to run many channels in parallel (35 channels in your case), and we are working on getting the CPU consumption, and resulting slowness of the JSON-RPC interface under control.

This seems to be a duplicate of #796
